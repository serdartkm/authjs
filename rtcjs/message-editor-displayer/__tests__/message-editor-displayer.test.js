

import { h } from 'preact'
import { render, fireEvent } from '@testing-library/preact'
import renderer from 'preact-render-to-string'
import MessageEditorDisplayer from '../index'

import '@testing-library/jest-dom/extend-expect'
describe('MessageEditorDisplayer component', () => {


    it("Should render correctly", () => {

        const tree = renderer(MessageEditorDisplayer)
        expect(tree).toMatchSnapshot();
    })

    it.only("Send button clicked once",()=>{
    let spy =jest.fn()

    const { debug,getByTestId,getByPlaceholderText } = render( <MessageEditorDisplayer id={0} sendMessage={spy} />,{})
     expect(getByTestId('sendMessage0')).toBeInTheDocument()
     expect(getByPlaceholderText('//i'))
    //  expect(wrapper.find('.btn')).toHaveLength(1)
    //   wrapper.find('.btn').simulate('click')
    //   expect(spy).toHaveBeenCalled()
    })


})
