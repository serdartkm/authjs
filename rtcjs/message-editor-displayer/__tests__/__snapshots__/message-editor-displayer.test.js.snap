// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageEditorDisplayer component Should render correctly 1`] = `
"function MessageEditorDisplayer({
  onMessageChange,
  message,
  sendMessage,
  style,
  id =
  /* istanbul ignore next */
  (cov_z2x33qnza.b[0][0]++, 0),
  disabled
}) {
  /* istanbul ignore next */
  cov_z2x33qnza.f[0]++;
  cov_z2x33qnza.s[0]++;
  return (
    /* istanbul ignore next */
    React.createElement(
    /* istanbul ignore next */
    &quot;div&quot;,
    /* istanbul ignore next */
    {
      style: { ...style,
        display: &quot;flex&quot;
      }
    },
    /* istanbul ignore next */
    React.createElement(
    /* istanbul ignore next */
    &quot;input&quot;,
    /* istanbul ignore next */
    {
      style: {
        flex: 1,
        padding: 5
      },

      /* istanbul ignore next */
      &quot;data-testid&quot;: \`message\${id}\`,
      onKeyDown: e =&gt; {
        /* istanbul ignore next */
        cov_z2x33qnza.f[1]++;
        cov_z2x33qnza.s[1]++;
        e.keyCode === 13 ?
        /* istanbul ignore next */
        (cov_z2x33qnza.b[1][0]++, sendMesage()) :
        /* istanbul ignore next */
        (cov_z2x33qnza.b[1][1]++, null);
      },
      className: &quot;form-control&quot;,
      onChange: onMessageChange,
      value: message,
      name: &quot;message&quot;,
      type: &quot;text&quot;,
      placeholder: &quot;Enter message text&quot;
    }),
    /* istanbul ignore next */
    React.createElement(
    /* istanbul ignore next */
    &quot;div&quot;,
    /* istanbul ignore next */
    {
      style: {
        width: 100
      }
    },
    /* istanbul ignore next */
    React.createElement(
    /* istanbul ignore next */
    &quot;button&quot;,
    /* istanbul ignore next */
    {
      /* istanbul ignore next */
      &quot;data-testid&quot;: \`sendMessage\${id}\`,
      disabled:
      /* istanbul ignore next */
      (cov_z2x33qnza.b[2][0]++, message === &quot;&quot;) ||
      /* istanbul ignore next */
      (cov_z2x33qnza.b[2][1]++, disabled),
      style: {
        marginLeft: 2,
        padding: 5,
        width: &quot;100%&quot;
      },
      onClick: sendMessage
    },
    /* istanbul ignore next */
    &quot;Send&quot;)))
  );
}"
`;
